File: Data Structures & Algorithms/remove-duplicates-from-sorted-array/submission-2.java
class Solution {
    public int removeDuplicates(int[] nums) {
        int p1 = 1;
        for (int p2 = 1; p2 < nums.length; p2++) {
            if (nums[p2] != nums[p2 - 1]) {
                nums[p1] = nums[p2];
                p1++;
            }
        }
        return p1;
    }
}

File: Data Structures & Algorithms/max-consecutive-ones/submission-2.py
class Solution:
    def findMaxConsecutiveOnes(self, nums: List[int]) -> int:
        maxOnes = currMax = 0
        for num in nums:
            if num == 0:
                maxOnes = max(currMax, maxOnes)
                currMax = 0
            else:
                currMax += 1
        
        return max(currMax, maxOnes)
            